src/main/java/sirius/biz/tenants/deletion/DeleteCodeListsWorkItem.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.tenants.deletion;
+
+import sirius.biz.codelists.CodeList;
+import sirius.biz.codelists.CodeListEntry;
+import sirius.biz.codelists.CodeLists;
+import sirius.biz.process.logs.ProcessLog;
+import sirius.biz.tenants.Tenant;
+import sirius.db.mixing.BaseEntity;
+import sirius.db.mixing.BaseMapper;
+import sirius.db.mixing.Mixing;
+import sirius.db.mixing.query.Query;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.Register;
+
+/**
+ * Deletes the {@link CodeList codeLists} and {@link CodeListEntry codeListEntries} of the given tenant.
+ */
+@Register
+public class DeleteCodeListsWorkItem implements TenantDeletionWorkItem {
```

Comment:
wouldn't this code be much more readable if we'd provide a concrete impl for mongo & sql?